src/markdown_parser.py

```diff
@@ -51,9 +51,22 @@ def urlreplace(matchobj: Match[str]) -> str:
 
         return re.sub(URL_REGEX, urlreplace, text)
 
+    def link(self, link, text=None, title=None):
+        is_asana_vanity_link = (
+            link.startswith("https://app.asana.com") and text is not None
+        )
+        if text is None:
+            text = link
+
+        asana_tags = 'data-asana-dynamic="false" ' if is_asana_vanity_link else ""
+        s = "<a " + asana_tags + 'href="' + self._safe_url(link) + '"'
+        if title:
+            s += ' title="' + mistune.escape_html(title) + '"'
+        return s + ">" + (text or link) + "</a>"
+
     # Asana's API can't handle img tags
     def image(self, src, alt="", title=None) -> str:
```

Comment:
Since your `link` function expects `None` as a default value, I think you should (if possible) change this to use `alt=None` instead of `alt=""`. The way you have it, if someone calls `image(src)` won't they end up with `<a data-asana-dynamic="false"></a>"`?

Reply:
Hm yeah this is odd - for reference, I'm just overriding the mistune functions (https://github.com/lepture/mistune/blob/master/mistune/renderers.py#L137-L146), so the only code calling these functions are the markdown parser itself. I think I'd like to keep the function signature the same, but instead in `def link` assume vanity Asana URL if text is not None OR text is not ""

Reply:
That sounds good — or just do a truthy/falsy check, `if text`